userspace/libscap/linux/scap_userlist.c

```diff
@@ -197,12 +197,14 @@ int32_t scap_linux_create_userlist(struct scap_platform* platform)
 		endpwent();
 	}
 
+	// if userIdx == 0 -> realloc with size 0 means free, and NULL is returned.
+	// so, we will end up with userlist->nusers = 0 and userlist->users NULL.
 	userlist->nusers = useridx;
 	if (useridx < usercnt)
 	{
 		// Reduce array size
 		scap_userinfo *reduced_userinfos = realloc(userlist->users, useridx * sizeof(scap_userinfo));
-		if(reduced_userinfos == NULL)
+		if(reduced_userinfos == NULL && useridx > 0)
```

Comment:
I see that there are other `free(userlist->users);` in this function, is it ok to continue the flow with `userlist->users == null` and `useridx==0`? can we interrupt the flow here if we follow in this condition ?

Reply:
It's not an error; it just means that there are no users/groups available. 
I think it's a super edge case, but still.
Freeing a NULL ptr it's not an error.

Reply:
ok thank you!